cg/apps/madeline.py

```diff
@@ -1,65 +1,76 @@
-# -*- coding: utf-8 -*-
-import os.path
-import subprocess
+"""Code to handle madeline input and output"""
+import logging
 from tempfile import NamedTemporaryFile, mkdtemp
 from typing import List
 
-SEX_GENDER = {'male': 'M', 'female': 'F'}
-STATUS_AFFECTED = {'affected': 'Y', 'unaffected': 'N'}
-COLUMNS = {
-    'family': 'FamilyId',
-    'sample': 'IndividualId',
-    'sex': 'Gender',
-    'father': 'Father',
-    'mother': 'Mother',
-    'deceased': 'Deceased',
-    'proband': 'Proband',
-    'status': 'Affected',
-}
+from cg.utils import Process
+
+LOG = logging.getLogger(__name__)
 
 
 def make_ped(family_id: str, samples: List[dict]):
-    """Make Madeline pedigree file."""
-    yield '\t'.join(COLUMNS.values())
+    """Yield lines that are used as madeline input."""
+    columns = {
+        "family": "FamilyId",
+        "sample": "IndividualId",
+        "sex": "Gender",
+        "father": "Father",
+        "mother": "Mother",
+        "deceased": "Deceased",
+        "proband": "Proband",
+        "status": "Affected",
+    }
+    sex_gender = {"male": "M", "female": "F"}
+    status_affected = {"affected": "Y", "unaffected": "N"}
+    LOG.info("Generating madeline input lines")
+
+    yield "\t".join(columns.values())
+
     for sample in samples:
         row = [
             family_id,
-            sample['sample'],
-            SEX_GENDER.get(sample['sex'], '.'),
-            sample.get('father') or '.',
-            sample.get('mother') or '.',
-            'Y' if sample.get('deceased') else '.',
-            'Y' if sample.get('proband') else '.',
-            STATUS_AFFECTED.get(sample.get('status'), '.'),
+            sample["sample"],
+            sex_gender.get(sample["sex"], "."),
+            sample.get("father") or ".",
+            sample.get("mother") or ".",
+            "Y" if sample.get("deceased") else ".",
+            "Y" if sample.get("proband") else ".",
+            status_affected.get(sample.get("status"), "."),
         ]
-        yield '\t'.join(row)
+        yield "\t".join(row)
 
 
-def run(executable: str, ped_stream: List[str]):
+def run(madeline_process: Process, ped_stream: List[str]):
```

Comment:
This could be a method in a class, where ```madeline_process``` is an attribute of type ```Process```, as we have done with other tools.

Reply:
Yes!